x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/KqlPlugin.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.kql;
+
+import org.elasticsearch.plugins.ExtensiblePlugin;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.plugins.SearchPlugin;
+
+public class KqlPlugin extends Plugin implements SearchPlugin, ExtensiblePlugin {
```

Comment:
`SearchPlugin` is present because the plugin will ship a new query.

Also `ExtensiblePlugin` has been added so we can have another plugin that depends of this one.
When we will be adding it as a dependency of the esql plugin we will not have this error message:

```
fatal exception while booting Elasticsearch java.lang.IllegalStateException: Plugin [x-pack-esql] cannot extend non-extensible plugin [x-pack-kql]
```

Maybe it is a little bit early, but I would prefer to keep it here since it is 100% sure that we will add it.

Reply:
I'd say let's implement them as we need'em - but I'm ok with this